File: src/main/java/net/csibio/aird/compressor/bytes/SnappyWrapper.java
package net.csibio.aird.compressor.bytes;

import java.io.IOException;
import java.util.Arrays;

import net.csibio.aird.enums.ByteCompType;
import org.xerial.snappy.Snappy;

public class SnappyWrapper implements ByteComp {

  @Override
  public String getName() {
    return ByteCompType.Snappy.getName();
  }

  @Override
  public byte[] encode(byte[] input) {
    try {
      return Snappy.compress(input);
    } catch (IOException e) {
      e.printStackTrace();
    }
    return null;
  }

  @Override
  public byte[] decode(byte[] input) {
    try {
      return Snappy.uncompress(input);
    } catch (IOException e) {
      e.printStackTrace();
    }
    return null;
  }

  @Override
  public byte[] decode(byte[] input, int offset, int length) {
    return decode(Arrays.copyOfRange(input, offset, offset + length));
  }
}

File: PythonSDK/Compressor/ByteComp/SnappyWrapper.py
from Compressor.ByteComp import ByteComp
from Enums.ByteCompType import ByteCompType
import snappy


class SnappyWrapper(ByteComp):

    def getName(self):
        return ByteCompType.Snappy

    def encode(self, input):
        return snappy.compress(input)

    def decode(self, input):
        return snappy.decompress(input)

    def decode(self, input, offset, length):
        return snappy.decompress(input[offset, offset + length])
